nservicebus/sqlserver/deployment-options.md

```diff
@@ -0,0 +1,61 @@
+tags:
+- SQL Server
+- Transports
+redirects:
+- nservicebus/sqlserver/multiple-databases
+---
+
+In default configuration SQL Server transport assumes that all tables used for storing messages are located in a single catalog and within a single schema. The configuration can be changed to partition message storage between different schemas and catalogs. The decision can me made on per queue basis (including error and audit queues).
+
+The supported deployment options are:
+ * ***default***: all queues are stored in a single catalog and a single schema (Versions 1.x and higher)
+ * ***multi-schema***: queues are stored in a single catalog but in more than one schema (Versions 2.1 and higher)
+ * ***multi-catalog***: queues are stored in multiple catalogs but on a single SQL Server instance. This mode is indirectly supported by using *multi-instance* option, and requires using DTC. In this document both options are covered under *multi-instance* term.
+ * ***multi-instance***: queues are stored in multiple catalogs on more than one SQL Server instance (Versions 2.1 to 2.x).
+
+NOTE: To properly identify the chosen deployment option all queues that the endpoint interacts with need to be taken into consideration, including error and audit queues. If either of them are stored in a separate SQL Server instance then the deployment is a *multi-instance* one.
+
+The transport will route messages to destination based on the configuration. If no specific configuration has been provided for a particular destination, the transport assumes the destination has the same configuration as the sending endpoint (i.e. identical schema, catalog and instance name). If the destination has a different configuration and it hasn't been provided, then exception will be thrown immediately, because the transport wouldn't be able to connect to the destination queue.
+
+# Modes overview
+
+## Default (single schema, single catalog, single SQL Server instance)
+
+- Supported in Versions 1.0 and higher.
+- Has simple configuration and setup.
+- Doesn't require Distributed Transaction Coordinator (MS DTC).
+- The snapshot (backup) of the entire system state can be done by backing up a single database. It is especially useful if business data is also stored in the same database.
+- Can be monitored with ServiceControl.
+
+## Multi-schema
+
+- Supported in Versions 2.1 and higher.
+- Has simple configuration and setup.
+- Doesn't require Distributed Transaction Coordinator (MS DTC).
+- The snapshot (backup) of the entire system state is done by backing up a single database. It is especially useful if business data is also stored in the same database.
+- Enables security configuration on a schema level.
+- Can't be monitored with ServiceControl.
+
+## Multi-instance (and multi-catalog)
+
+- Supported in Versions 2.1 to 2.x, legacy support is offered in Versions 3.x, won't be supported directly in Versions 4.0 and higher.
```

Comment:
I think we should add:
WARNING: This option will not be supported in Versions 4 and above.